sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -0,0 +1,205 @@
+        plot_method : {'contourf', 'contour', 'pcolormesh'}, default='contourf'
+            Plotting method to call when plotting the response. Please refer
+            to the following matplotlib documentation for details:
+            :func:`contourf <matplotlib.pyplot.contourf>`,
+            :func:`contour <matplotlib.pyplot.contour>`,
+            :func:`pcolomesh <matplotlib.pyplot.pcolomesh>`.
+
+        ax : Matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
+
+        **kwargs : dict
+            Additional keyword arguments to be pased to the `plot_method`.
+
+        Returns
+        -------
+        display: :class:`~sklearn.inspection.DecisionBoundaryDisplay`
+        """
+        check_matplotlib_support('DecisionBoundaryDisplay.plot')
+        import matplotlib.pyplot as plt  # noqa
+
+        if plot_method not in ('contourf', 'contour', 'pcolormesh'):
+            raise ValueError("plot_method must be 'contourf', 'contour', or "
+                             "'pcolormesh'")
+
+        if ax is None:
+            _, ax = plt.subplots()
+
+        plot_func = getattr(ax, plot_method)
+        self.surface_ = plot_func(self.xx0, self.xx1, self.response, **kwargs)
+        self.ax_ = ax
+        self.figure_ = ax.figure
+        return self
+
+
+def plot_decision_boundary(estimator, X, grid_resolution=100, eps=1.0,
+                           plot_method='contourf', response_method='auto',
+                           ax=None, **kwargs):
+    """Plot Decision Boundary.
+
+    Please see examples below for usage.
+
+    Parameters
+    ----------
+    estimator : estimator instance
+        Trained estimator.
+
+    X : array-like of shape (n_samples, 2)
+        Input values.
+
+    grid_resolution : int, default=100
+        The number of equally spaced points to evaluate the response function.
+
+    eps : float, default=1.0
+        Extends the minimum and maximum values of X for evaluating the
+        response function.
+
+    plot_method : {'contourf', 'contour', 'pcolormesh'}, default='contourf'
+        Plotting method to call when plotting the response. Please refer
+        to the following matplotlib documentation for details:
+        :func:`contourf <matplotlib.pyplot.contourf>`,
+        :func:`contour <matplotlib.pyplot.contour>`,
+        :func:`pcolomesh <matplotlib.pyplot.pcolomesh>`.
+
+    response_method : {'auto', 'predict_proba', 'decision_function', \
+            'predict'}, defaul='auto'
+        Specifies whether to use :term:`predict_proba`,
+        :term:`decision_function`, :term:`predict` as the target response.
+        If set to 'auto', the response method is tried in the following order:
+        :term:`predict_proba`, :term:`decision_function`, :term:`predict`.
+
+    ax : Matplotlib axes, default=None
+        Axes object to plot on. If `None`, a new figure and axes is
+        created.
+
+    **kwargs : dict
```

Comment:
I am also thinking that we should have a way to decorate xlabel and ylabel if we are passing a dataframe using the column names.